main.go

```diff
@@ -123,11 +127,39 @@ func main() {
 	// Namespace where the operator is installed
 	currentNamespace := env.ReadOrPanic(util.CurrentNamespace)
 
+	// Get trace and span IDs from environment variables
+	traceIDHex := os.Getenv("OTEL_TRACE_ID")
+	spanIDHex := os.Getenv("OTEL_PARENT_ID")
+	endpoint := os.Getenv("OTEL_EXPORTER_OTLP_ENDPOINT")
+
 	// Get a config to talk to the apiserver
 	cfg := ctrl.GetConfigOrDie()
 
+	log.Debugf("Setting up tracing with ID: %s, Parent ID: %s, Endpoint: %s", traceIDHex, spanIDHex, endpoint)
+	traceCtxWithParentSpan, tp, err := telemetry.SetupTracingFromParent(signalCtx, traceIDHex, spanIDHex, endpoint)
```

Comment:
can we simplify this? 
We don't need to create new var names for each span - we can reuse ctx. Methods inside are returning ctx if err, so we don't lose anything if error happens.